Credential card rework.

JoshDunnDev opened the pull request.

Commit:
Change of layout of card.

Adds icons for each field. Removes show/share buttons can toggle details section. Limits the field maximum to 3 fields.

Commit:
Add detail mode and finish card layout.

Commit:
Refactor logic with mixin.

Commit:
Clean up of CSS for Credential components.

Commit:
Add functionality for sublabel fields.

mattcollier commented on CredentialCardDetail.vue:
```diff
@@ -0,0 +1,174 @@
+      <div class="column items-center s-card-info-left">
+        <q-card-media>
+          <q-icon
+            v-if="useDefaultImage || !credential.issuerLogo & !credential.image"
```
I don't think this does the right thing, see: https://repl.it/repls/GuiltySpotlessScans

JoshDunnDev replied:
Updated.

mattcollier commented on CredentialCardDetail.vue:
```diff
@@ -0,0 +1,174 @@
+</template>
+<script>
+/*!
+ * Copyright (c) 2018 Digital Bazaar, Inc. All rights reserved.
```
Date here and throughout.

JoshDunnDev replied:
Updated.

mattcollier commented on credentialMixin.js:
```diff
@@ -0,0 +1,79 @@
+  },
+  computed: {
+    fields() {
+      if(!this.credential || !this.schema) {
```
preferred: `if(!(a && b))`

JoshDunnDev replied:
Updated.

mattcollier commented on credentialMixin.js:
```diff
@@ -0,0 +1,79 @@
+};
+
+function _createFields(fields, source, schema) {
+  console.log('SOURCE', source);
```
remove?

JoshDunnDev replied:
Removed.

mattcollier commented on credentialMixin.js:
```diff
@@ -0,0 +1,79 @@
+  console.log('SOURCE', source);
+  for(const key in source) {
+    // naively recurse into objects
+    if(typeof source[key] === 'object') {
```
Does this behave as expected if `source[key]` is an array (which also has typeof object).

mattcollier replied:
Seems like this sort of algo has been developed a number of times, maybe there's something we can use?
@gannan08 @dlongley @davidlehn

mattcollier reviewed (changes requested):
done with first pass.

Commit:
Refactor if statements and change dates.

Also removed console.log left in credentialMixin.

JoshDunnDev commented:
@mattcollier Using `!(credential.issuerLogo && credential.image)` appears to not be working the same as `!credential.issuerLogo && !credential.image`.

Commit:
Refactor of v-if statement.

Commit:
Refactor of v-if for CredentialCardField.

mattcollier reviewed (approved):
Only open question is related to recursive `_createFields` helper which I will leave to others to sort out.

Commit:
Refactor to fix max length in field component.

Commit:
Change background color of card to white.

Commit:
Add 2 line support for CredentialCard title.

Commit:
Add multiline support for long values.

Commit:
Add hover to icons to show field name.

Commit:
Change default image to allow manual set.

Can specify default image in attribute of credential.

Commit:
Small change to CSS for icon size.

Commit:
Change to keep original default image logic.

Commit:
Remove unnecessary click event on card.

Commit:
Add list view for CHAPI.

davidlehn commented on CredentialCardList.vue:
```diff
@@ -0,0 +1,134 @@
+      <q-card-separator />
+    </div>
+
+    <div v-if="showFieldValues === true">
```
Can this just test for truthiness?
```suggestion
    <div v-if="showFieldValues">
```

Commit:
Change logic for showFieldValues.

Commit:
Change to CSS for CredentialCardList for mobile.

aljones15 committed:
Add eslint to project. (#2)

* Add eslint to project.

Add all eslintrc.js.

Eslint fix ran.

Add babel-eslint to test.

Finish implementing eslint.

* Add start script.

* Remove all lint errors from components.

gannan08 commented:
Closed in preference of https://github.com/digitalbazaar/bedrock-vue-credential-card/pull/5.
